src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -215,43 +215,44 @@ uct_ib_mlx5_set_dgram_seg(struct mlx5_wqe_datagram_seg *seg,
 static UCS_F_ALWAYS_INLINE void
 uct_ib_mlx5_set_ctrl_seg(struct mlx5_wqe_ctrl_seg* ctrl, uint16_t pi,
                          uint8_t opcode, uint8_t opmod, uint32_t qp_num,
-                         uint8_t fm_ce_se, unsigned wqe_size)
+                         uint8_t fm_ce_se, unsigned wqe_size, uint32_t imm)
 {
     uint8_t ds;
 
     ucs_assert(((unsigned long)ctrl % UCT_IB_MLX5_WQE_SEG_SIZE) == 0);
     ds = ucs_div_round_up(wqe_size, UCT_IB_MLX5_WQE_SEG_SIZE);
 #if defined(__SSE4_2__)
     *(__m128i *) ctrl = _mm_shuffle_epi8(
-                    _mm_set_epi32(qp_num, ds, pi,
+                    _mm_set_epi32(qp_num, imm, (ds << 16) | pi,
```

Comment:
can you please check if this has an effect on UCT message rate/latency? (even with FC=off)
since this change adds some overhead
maybe we can check in this function if the opcode is IMM and revert to previous behavior if not (this check would be resolved in compile-time since opcode is constant)